ompi/mpi/c/alltoallw_init.c

```diff
@@ -127,7 +127,7 @@ int MPI_Alltoallw_init(const void *sendbuf, const int sendcounts[], const int sd
                                             rdispls, recvtypes, comm, info, request,
                                             comm->c_coll->coll_alltoallw_init_module);
     if (OPAL_LIKELY(OMPI_SUCCESS == err)) {
-        ompi_coll_base_retain_datatypes_w(*request, (MPI_IN_PLACE==sendbuf)?NULL:sendtypes, recvtypes);
+        ompi_coll_base_retain_datatypes_w(*request, (MPI_IN_PLACE==sendbuf)?NULL:sendtypes, recvtypes, true);
```

Comment:
Should this one be "false" since it's a regular alltoallw instead of a neighborhood collective?  If I'm understanding it right, we'd only want the counts to be based on mca_topo_base_neighbor_count() if we're in a neighborhood collective, so here we'd still want the counts to be based on ompi_comm_size()

Reply:
good catch.